### DISCUSSION TITLE
Dragging an object with SkiaSharp

### DISCUSSION THREAD

**Discussion by AghaKhan** [mono/SkiaSharp#2632] [Q&A]
### Description

I wanted to develop a game where I would like to use SikaSharp for the Mobile platform. The provided samples only work for UWP. Even the models on UWP are complex because one sample code implements more than one concept and makes the code useless. Too much information is no information.

### Code

Just need simple dragging with Skiashap which can run all devices.

### Version of SkiaSharp

2.88.3 (Current)

### Last Known Good Version of SkiaSharp

2.88.2 (Previous)

### IDE / Editor

Visual Studio (Windows)

### Platform / Operating System

All

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by mattleibow** [comment]
SkiaSharp is more of a base level library in that it does not support many abstract actions - such as a concept of dragging or clicking.

If you want to support dragging, you will have to do most of the work yourself:
 - keeping track of objects to drag
 - keeping track of the tap up/down/move events
 - moving the object by updating coordinates
 - supporting any cancellation or dropping actions

I have the most basic sample where I "drag" text around. Not really dragging as the text snaps to the tap location, but this can be changed to first detect tap location: https://github.com/mono/SkiaSharp/blob/main/samples/Basic/Maui/SkiaSharpSample/MainPage.xaml.cs

**Comment by AghaKhan** [comment]
@mattleibow
I was able to implement dragging with the PinchPanContainer sample. All the pieces written for UWP must be rewritten for Android. After looking at those samples, I should be able to rewrite them for all devices.
